lib/spack/spack/repo.py

```diff
@@ -1164,14 +1164,14 @@ def create_or_construct(path, namespace=None):
     return Repo(path)
 
 
-def _path():
+def _path(repo_dirs=None):
```

Comment:
Is this just to make it easier to patch this for testing?

Reply:
The change is for tests, but I would say that it is necessary, since I need to transmit the in-memory state of the test process to the subprocess, and without this change the `RepoPath` is only constructed from disk (e.g. bypassing the mock repo established for the test).